core/slim_globals.h

```diff
@@ -354,6 +354,432 @@ extern EidosValue_String_SP gStaticEidosValue_StringT;
+"        \"genome_type\": {\n"
+"            \"binaryFormat\": \"B\",\n"
+"            \"description\": \"The 'type' of this genome (autosome, sex chromosome, etc).\",\n"
+"            \"index\": 2,\n"
+"            \"type\": \"integer\"\n"
+"        },\n"
+"        \"is_null\": {\n"
+"            \"binaryFormat\": \"B\",\n"
+"            \"description\": \"Whether this node describes a 'null' (non-existant) chromosome.\",\n"
+"            \"index\": 1,\n"
+"            \"type\": \"integer\"\n"
+"        },\n"
+"        \"slim_id\": {\n"
+"            \"binaryFormat\": \"q\",\n"
+"            \"description\": \"The 'pedigree ID' of this chromosome in SLiM.\",\n"
+"            \"index\": 0,\n"
+"            \"type\": \"integer\"\n"
+"        }\n"
+"    },\n"
+"    \"required\": [\n"
+"        \"slim_id\",\n"
+"        \"is_null\",\n"
+"        \"genome_type\"\n"
+"    ],\n"
+"    \"type\": [\n"
+"        \"object\",\n"
+"        \"null\"\n"
+"    ]\n"
+"}";
+
+const std::string SLIM_TSK_INDIVIDUAL_METADATA_SCHEMA =
+"{\n"
+"    \"$schema\": \"http://json-schema.org/schema#\",\n"
+"    \"additionalProperties\": false,\n"
+"    \"codec\": \"struct\",\n"
+"    \"description\": \"SLiM schema for individual metadata.\",\n"
+"    \"examples\": [\n"
+"        {\n"
+"            \"age\": -1,\n"
+"            \"flags\": 0,\n"
+"            \"pedigree_id\": 123,\n"
+"            \"population\": 0,\n"
+"            \"sex\": 0\n"
+"        }\n"
+"    ],\n"
+"    \"flags\": {\n"
+"        \"SLIM_INDIVIDUAL_METADATA_MIGRATED\": {\n"
+"            \"description\": \"Whether this individual was a migrant, either in the generation when the tree sequence was written out (if the individual was alive then), or in the generation of the last time they were Remembered (if not).\",\n"
+"            \"value\": 1\n"
+"        }\n"
+"    },\n"
+"    \"properties\": {\n"
+"        \"age\": {\n"
+"            \"binaryFormat\": \"i\",\n"
+"            \"description\": \"The age of this individual, either when the tree sequence was written out (if the individual was alive then), or the last time they were Remembered (if not).\",\n"
+"            \"index\": 2,\n"
+"            \"type\": \"integer\"\n"
+"        },\n"
+"        \"flags\": {\n"
+"            \"binaryFormat\": \"I\",\n"
+"            \"description\": \"Other information about the individual: see 'flags'.\",\n"
+"            \"index\": 5,\n"
+"            \"type\": \"integer\"\n"
+"        },\n"
+"        \"pedigree_id\": {\n"
+"            \"binaryFormat\": \"q\",\n"
+"            \"description\": \"The 'pedigree ID' of this individual in SLiM.\",\n"
+"            \"index\": 1,\n"
+"            \"type\": \"integer\"\n"
+"        },\n"
+"        \"population\": {\n"
+"            \"binaryFormat\": \"i\",\n"
+"            \"description\": \"The population the individual was part of, either when the tree sequence was written out (if the individual was alive then), or the last time they were Remembered (if not).\",\n"
```

Comment:
I'll change this in the descriptions, but not the names of the keys, I think? Because (a) it's been like this for a while in pyslim, and changing it would break things, and (b) these are actually the IDs of tskit populations, eg if it has SLiM subpopulation 3, that refers to `ts.population(3)` - usually, anyhow.

Reply:
Hmm.  Well, breaking things is bad.  But is (b) true?  Aren't they SLiM subpopulation IDs, not tskit population IDs?  Or are the numbers getting translated somewhere by code I'm not aware of?  Or by "usually, anyhow" do you mean that they just happen to coincidentally line up in typical usage cases?  If so, that only makes the potential for confusion greater, and the need for clarity and consistency in terminology greater, to me.  Hmm.

Reply:
Well, the IDs match for the subpops that are around at the end of the simulation. If users have removed populations along the way, these might indeed not match. We've talked about doing this differently, but decided that it's not worth the trouble.

Actually, we could change this name to `subpopulation` now and not break things. (However, it will still be visible in pyslim as e.g. `mutation.metadata.population`.) I will do that - it's better, since as you say, these are different things.

Reply:
> Well, the IDs match for the subpops that are around at the end of the simulation. If users have removed populations along the way, these might indeed not match. We've talked about doing this differently, but decided that it's not worth the trouble.

  But if a SLiM model defines a single subpopulation named p1000 instead of p1, what I'm asking is, is the id here 1000 or 1?  If it's the SLiM subpop id, then it's 1000, and will not match tskit.  If that's the case, then we should definitely use the "subpopulation" terminology; otherwise we're using tskit terminology to refer to SLiM subpop ids, which is bound to cause confusion somewhere down the line.

> Actually, we could change this name to `subpopulation` now and not break things. (However, it will still be visible in pyslim as e.g. `mutation.metadata.population`.) I will do that - it's better, since as you say, these are different things.

OK.  Since pyslim is all about translating SLiM things into tskit terms, that seems fine to me.  (But then I hope pyslim is actually vending the tskit population index, not the SLiM subpopulation id...?)

Reply:
The SLiM subpop ID matches the tskit population ID, so if you have a siingle subpop p1000 then the population table will have 1000 empty rows and then one row (the 1000th, zero-indexed) with the information about that population, and references to either slim subpop ID or tskit population ID of 1000 will refer to that population/subpopulation.

So: the slim subpop ID and tskit population ID always match. The only problem is that slim doesn't have a notion of "the subpop with that ID at that point in time".

Reply:
Ah yes, this rings a bell now.  OK, good.

Reply:
Have a look at the [pyslim code](https://github.com/tskit-dev/pyslim/pull/89/files#diff-e1abd0a63c6a0ff314e018d205d42ca4R17) to see if you like the edits?

Reply:
Ah, that's a lotta diffs, in a language I'm not fluent in.  Is there a specific bit you want me to look at?

Reply:
sorry, i tried to make a link directly to it.. but, the schemas, in `pyslim/slim_metadata.py`: https://github.com/tskit-dev/pyslim/pull/89/files#diff-e1abd0a63c6a0ff314e018d205d42ca4

Reply:
Ah, GH wasn't going to the linked diff because it was part of a "large diff" that was "not loaded by default".  I see, you're just pointing me to the more readable original version of the schemas?  No worries, I reviewed them in the SLiM code despite the diminished readability, I think...?